app/models/mixins/assignment_mixin.rb

```diff
@@ -97,12 +97,11 @@ def assignments
       Tag
         .includes(:taggings).references(:taggings)
         .where("taggings.taggable_type = ? and tags.name like ?", name, "#{namespace}/%")
-        .flat_map do |tag|
-          tag.taggings.map do |tagging|
-            {
-              :assigned    => assignment_map[tagging.taggable_id],
-              :assigned_to => Tag.filter_ns([tag], namespace).first
-            }
+        .each_with_object({}) do |tag, ret|
+          tag.taggings.each do |tagging|
+            tag_name = Tag.filter_ns([tag], namespace).first
+            taggable = assignment_map[tagging.taggable_id]
+            (ret[tag_name] ||= []) << taggable if taggable
```

Comment:
This can be simplified by initializing the Hash as `Hash.new { |h, k| h[k] = [] }`.  That way this line just becomes `ret[:tag_name] << taggable if taggable`.  Also it simplifies the other defaulting you are doing on line 129.  That line goes from `.flat_map { |t| alist[t] || [] }` to `.flat_map { |t| alist[t] }`